tests/phpunit/CRM/Pledge/BAO/PledgeTest.php

```diff
@@ -129,4 +129,59 @@ public function testRetrieveKnownPledgeID() {
     $this->assertEquals(count($pledgeId), 1, "Pledge was retrieved");
   }
 
+  /**
+   *  Test create recur record.
+   */
+  public function testCreateRecurRecord() {
+    //when user creating pledge record.
+    $contributionPageId = ContributionPage::create();
+    $processorId = $this->paymentProcessorAuthorizeNetCreate(array('is_test' => 0));
+    $contribution = $this->contributionCreate(array('contact_id' => $this->_contactId));
+    $pledgeParams = array(
+      'contact_id' => $this->_contactId,
+      'installment_amount' => 100,
+      'actual_amount' => 100,
+      'contribution_id' => $contribution,
+      'contribution_page_id' => $contributionPageId,
+      'financial_type_id' => 1,
+      'frequency_interval' => 1,
+      'installments' => 3,
+      'frequency_unit' => 'month',
+      'frequency_day' => intval(date("d")),
+      'start_date' => date("Ymd"),
+      'create_date' => date("Ymd"),
+      'scheduled_date' => date("Ymd"),
+      'status_id' => 1,
+      'max_reminders' => 1,
+      'initial_reminder_day' => 1,
+      'additional_reminder_day' => 1,
+      'original_installment_amount' => 100,
+    );
+
+    $pledge = CRM_Pledge_BAO_Pledge::create($pledgeParams);
```

Comment:
It's generally better to call $this->callAPISuccess - then the defaults kick in & you can shorten the array, you get better error reporting & you encourage people to call the api not the BAO